subgraphs/notional-finance/src/setters/financialMetrics.ts

```diff
@@ -29,52 +27,60 @@ import { getOrCreateToken } from "../getters/token";
 
 export function updateFinancials(
   event: ethereum.Event,
-  amountUSD: BigDecimal,
-  marketId: string
+  marketId: string,
+  avgInterestRate: BigDecimal
 ): void {
-  const financialsDailySnapshots = getOrCreateFinancialsDailySnapshot(event);
+  const market = getOrCreateMarket(event, marketId);
   const marketHourlySnapshot = getOrCreateMarketHourlySnapshot(event, marketId);
   const marketDailySnapshot = getOrCreateMarketDailySnapshot(event, marketId);
   const protocol = getOrCreateLendingProtocol();
-  const market = getOrCreateMarket(event, marketId);
+  const financialsDailySnapshots = getOrCreateFinancialsDailySnapshot(event);
+
+  // total/cumulative revenue calculation
+  const supplySideRevenueUSD = market.totalDepositBalanceUSD
```

Comment:
Just using average interest rate is over simplifying imo. I think we should take into account when the timestamp the market was opened and closed and normalize the rate to the time it was open.

I am actually thinking of a simpler solution. Let me know what you think. Each time you open a new market snapshot you load the previous market snapshot, then calculate the daily revenue and update it:

`newSupplySideRevenue` = `prevSnapshot.totalDepositUSD`*`prevSnapshot.lendingRate` / `365`

And then when we close the market we know that is the final market snapshot so we can dod the calculation for the last snapshot and do not have to do it retrospectively. cc @ishraq8

Reply:
> I think we should take into account when the timestamp the market was opened and closed and normalize the rate to the time it was open.

* This is still going to be theoretical since we don't have actual timestamp of the market open and close.
* You are suggesting, we would be able to normalize the metrics because we are able to calculate the revenue based on daily lending rates as opposed to avg lending rates. Am I understanding your proposal correctly?

> And then when we close the market we know that is the final market snapshot so we can dod the calculation for the last snapshot and do not have to do it retrospectively.
* How would we know that is the final market snapshot when we are closing the market?
* What does "dod the calculation" mean?
* We would still need to update market total deposits/borrows retrospectively. We aren't changing any updating any revenue metrics retrospectively.

Reply:
1. What I mean with the first statement: Every time `updateFinancials()` is called you are taking the deposit amount * an annual interest rate, but you are calling this multiple times. So it is over counting. In addition a market may only be open for 200 days. If that is the case you are still multiplying by an annual rate which is over counting.

- With this current solution I was suggesting that we should update the revenue on market close instead of each time we call `updateFinancials()` to prevent some of the double counting.
- The issue I see with that is that we are not really accounting how much liquidity was earning interest 10 days ago vs today.
- And that is why I suggested the alternate solution...

2. This solution is essentially calculating the daily interest each time a snapshot is complete (complete in the sense that a new day has begun).

- We know the final snapshot is taken when we change `market.isActive` to `false`. I am assuming that markets do not open again after that. Correct me if I am wrong.
- "dod" was a typo. I meant "do"
- Why would you need to update deposits/borrows retrospectively? The only thing I see being changed is: when you create a new market snapshot you grab the previous snapshot and make this calculation: `newSupplySideRevenue = newTotalRevenue = prevSnapshot.totalDepositUSD*prevSnapshot.lendingRate / 365`
- Are there limitations that prevent us from doing this? After thinking about this more technically that seems to be the most accurate way to get our "theoretical" revenue.

Reply:
>Every time updateFinancials() is called you are taking the deposit amount * an annual interest rate, but you are calling this multiple times. So it is over counting.

Still unclear why this would be over counting. We are calling it only once per (deposit) event.

> In addition a market may only be open for 200 days. If that is the case you are still multiplying by an annual rate which is over counting.

Agree on this, was exploring alternatives to manage this.

> 2. This solution is essentially calculating the daily interest each time a snapshot is complete.

I like this new solution. Don't see any limitations to this approach at this time.

> We know the final snapshot is taken when we change market.isActive to false. I am assuming that markets do not open again after that. Correct me if I am wrong.

Correct

> Why would you need to update deposits/borrows retrospectively?

* Not the deposits/borrows. The market metrics associated with deposits/borrows like deposit balance, borrow balance, tvl. This doesn't change any revenue metrics like I mentioned previously. 
* So, effectively, there will continue to be two changes with the new solution as well. 1) Update financials using totalRev and lendingRate from the prevSnapshot 2) updating market balances and position counts to 0.

Reply:
@dmelotik  Let me push the new solution and @ishraq8 can run a quick round of QA to see if that look's good. If good, we can go ahead and merge after code review is complete.